### ISSUE TITLE
secret key locked via `pgpainless-cli change-key-password` fails to sign

### ISSUE THREAD

**Issue by dkg** [vanitasvitae/pgpainless#453]
Using the versions in debian:

```
0 dkg@bob:~$ pgpainless-cli version --extended
PGPainless-SOP 1.6.7
https://codeberg.org/PGPainless/pgpainless/src/branch/master/pgpainless-sop

Implementation of the Stateless OpenPGP Protocol Version 07
https://datatracker.ietf.org/doc/html/draft-dkg-openpgp-stateless-cli-07

Based on pgpainless-core 1.6.7
https://pgpainless.org

Using Bouncy Castle 1.77
https://www.bouncycastle.org/java.html
0 dkg@bob:~$ pgpainless-cli generate-key 'Test <test@example.com>' > tmp/test.key
0 dkg@bob:~$ echo testpw > tmp/pw.txt
0 dkg@bob:~$ pgpainless-cli change-key-password --new-key-password=tmp/pw.txt < tmp/test.key > tmp/test-locked.key
0 dkg@bob:~$ echo sign | pgpainless-cli sign --with-key-password=tmp/pw.txt tmp/test-locked.key 
Key 856F0760353206E1A527B00D8EE3BB2F847CE730 cannot be unlocked.
67 dkg@bob:~$ 
```

Interestingly, i appear to be able to sign with a locked key if i generated the key using `pgpainless-cli generate-key --with-key-password`:

```
0 dkg@bob:~$ pgpainless-cli generate-key 'Locked <locked@example.com>' > tmp/test-original-lock.key
0 dkg@bob:~$ echo sign | pgpainless-cli sign --with-key-password=tmp/pw.txt tmp/test-original-lock.key 
-----BEGIN PGP SIGNATURE-----

iHUEABYKACcFAmchBX8JELHbE2hED3GnFiEEA0j9jdAwb5WH6PCrsdsTaEQPcacA
AHEtAP0ffRRjau8mAQzaWhzYCh7t073tYFJXpt+pPThrOHPQ5gD/Z4vzpuiI6mMH
0GNTgVbZMNcdzv+ZQpcpj1L+seQLogA=
=rBg+
-----END PGP SIGNATURE-----
0 dkg@bob:~$ 
```

**Comment by dkg** [comment]
interesting, i seem to see the same problem for `pgpainless-cli decrypt` and subsequent invocations of `pgpainless-cli change-key-password` when supplying `--with-key-password` for a `KEYS` object that has had its password changed with `change-key-password`.  Perhaps there's a common underlying issue.

**Comment by vanitasvitae** [comment]
Ah, the cause is that in `change-key-password`, the passphrase arguments are passed as direct, rather than indirect arguments. I'll fix that!

**Comment by vanitasvitae** [comment]
This is fixed in PGPainless 1.6.8 and 1.7.2

**Comment by vanitasvitae** [comment]
1.6.8 was uploaded to Debian unstable, so you might be able to test its correct functionality :)

**Comment by dkg** [comment]
thanks for this work!

but hm, i don't see it fixed:

```
0 dkg@bob:~$ pgpainless-cli version --extended
PGPainless-SOP 1.6.8
https://codeberg.org/PGPainless/pgpainless/src/branch/master/pgpainless-sop

Implementation of the Stateless OpenPGP Protocol Version 07
https://datatracker.ietf.org/doc/html/draft-dkg-openpgp-stateless-cli-07

Based on pgpainless-core 1.6.8
https://pgpainless.org

Using Bouncy Castle 1.77
https://www.bouncycastle.org/java.html
0 dkg@bob:~$ pgpainless-cli generate-key 'Test <test@example.com>' > tmp/test.key
0 dkg@bob:~$ echo testpw > tmp/pw.txt
0 dkg@bob:~$ pgpainless-cli change-key-password --new-key-password=tmp/pw.txt < tmp/test.key > tmp/test-locked
0 dkg@bob:~$ echo sign | pgpainless-cli sign --with-key-password=tmp/pw.txt tmp/test-locked.key
Key 856F0760353206E1A527B00D8EE3BB2F847CE730 cannot be unlocked.
67 dkg@bob:~$ 
```

Am i missing something?

**Comment by vanitasvitae** [comment]
Yeah, sop-java (where this bug was actually fixed) was not updated in Debian. The maintainer uploaded it today though, so it should soon be available :D

**Comment by dkg** [comment]
hm, maybe future versions of `pgpainless-cli version --extended` should also report the version of sop-java that is in use.  i've opened #454 to keep track of that.

**Comment by dkg** [comment]
btw, i can confirm that it is fixed, with sop-java 7.0.2 and pgpainless 1.6.8.  thank you!
